lib/src/components/segmented_control/segmented_control.dart

```diff
@@ -0,0 +1,632 @@
+  @override
+  Size computeDryLayout(BoxConstraints constraints) {
+    final Size childSize = _calculateChildSize(constraints);
+    return _computeOverallSizeFromChildSize(childSize, constraints);
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    RenderBox? child = firstChild;
+    double maxMaxChildHeight = ZetaSpacing.x7;
+    while (child != null) {
+      final double childHeight = child.getMaxIntrinsicHeight(width);
+      maxMaxChildHeight = math.max(maxMaxChildHeight, childHeight);
+      child = nonSeparatorChildAfter(child);
+    }
+    return maxMaxChildHeight;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    final int childCount = this.childCount ~/ 2 + 1;
+    RenderBox? child = firstChild;
+    double maxMaxChildWidth = 0;
+    while (child != null) {
+      final double childWidth = child.getMaxIntrinsicWidth(height);
+      maxMaxChildWidth = math.max(maxMaxChildWidth, childWidth);
+      child = nonSeparatorChildAfter(child);
+    }
+    return maxMaxChildWidth * childCount + totalSeparatorWidth;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    RenderBox? child = firstChild;
+    double maxMinChildHeight = ZetaSpacing.x7;
+    while (child != null) {
+      final double childHeight = child.getMinIntrinsicHeight(width);
+      maxMinChildHeight = math.max(maxMinChildHeight, childHeight);
+      child = nonSeparatorChildAfter(child);
+    }
+    return maxMinChildHeight;
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    final int childCount = this.childCount ~/ 2 + 1;
+    RenderBox? child = firstChild;
+    double maxMinChildWidth = 0;
+    while (child != null) {
+      final double childWidth = child.getMinIntrinsicWidth(height);
+      maxMinChildWidth = math.max(maxMinChildWidth, childWidth);
+      child = nonSeparatorChildAfter(child);
+    }
+    return (maxMinChildWidth + 2 * ZetaSpacing.l) * childCount + totalSeparatorWidth;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty<Rect?>('currentThumbRect', currentThumbRect))
+      ..add(DiagnosticsProperty<bool>('rounded', rounded))
+      ..add(DiagnosticsProperty<Paint>('separatorPaint', separatorPaint))
+      ..add(DiagnosticsProperty<_ZetaSegmentedControlState<T>>('state', state))
+      ..add(IntProperty('highlightedIndex', highlightedIndex))
+      ..add(ColorProperty('thumbColor', thumbColor))
+      ..add(DoubleProperty('totalSeparatorWidth', totalSeparatorWidth));
+  }
+
+  @override
+  void detach() {
+    state._thumbController.removeListener(markNeedsPaint);
+    super.detach();
+  }
+
+  @override
+  void handleEvent(PointerEvent event, BoxHitTestEntry entry) {
+    if (event is PointerDownEvent) state._tap.addPointer(event);
+  }
+
+  @override
+  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
```

Comment:
I don't know if this is the right place to comment this, but I think the hit area is too small for the taps - when we tap near the value I think it should be selected

https://github.com/zebrafed/zeta_flutter/assets/23633655/a9470d79-83ff-4195-95c1-a2aa48ae086f